Web/Edubase.Web.UIUnitTests/Areas/ViewRulesHandlers/GroupEditorViewModelRulesHandlerTests.cs

```diff
@@ -0,0 +1,253 @@
+        [InlineData(eLookupGroupType.Federation, eLookupGroupStatus.CreatedInError, EdubaseRoles.YCS)]
+        [InlineData(eLookupGroupType.Federation, eLookupGroupStatus.Closed, EdubaseRoles.YCS)]
+        public void SetEditPermissions_GroupViewModel_CanUserCloseAndMarkAsCreatedInError_False(eLookupGroupType groupType, eLookupGroupStatus statusId, string role)
+        {
+            mockPrincipal.Setup(x => x.IsInRole(It.IsAny<string>())).Returns(false);
+            mockPrincipal.Setup(x => x.IsInRole(role)).Returns(true);
+            var viewModel = new GroupEditorViewModel()
+            {
+                GroupTypeId = (int) groupType,
+                StatusId = (int) statusId
+            };
+
+            var result = GroupEditorViewModelRulesHandler.SetEditPermissions(viewModel, mockPrincipal.Object);
+
+            Assert.False(result.CanUserCloseAndMarkAsCreatedInError);
+        }
+
+        [Theory]
+        [InlineData((int) eLookupGroupType.ChildrensCentresCollaboration, EdubaseRoles.ROLE_BACKOFFICE)]
+        [InlineData((int) eLookupGroupType.ChildrensCentresGroup, EdubaseRoles.ROLE_BACKOFFICE)]
+        public void SetEditPermissions_GroupViewModel_IsLocalAuthorityEditable_True(int groupType, string role)
+        {
+            mockPrincipal.Setup(x => x.IsInRole(It.IsAny<string>())).Returns(false);
+            mockPrincipal.Setup(x => x.IsInRole(role)).Returns(true);
+            var viewModel = new GroupEditorViewModel()
+            {
+                GroupTypeId = groupType,
+                LinkedEstablishments = new GroupLinkedEstablishmentsViewModel() { Establishments = new List<EstablishmentGroupViewModel>() }
+            };
+
+            var result = GroupEditorViewModelRulesHandler.SetEditPermissions(viewModel, mockPrincipal.Object);
+
+            Assert.True(result.IsLocalAuthorityEditable);
+        }
+
+        [Theory]
+        [InlineData((int) eLookupGroupType.SecureSingleAcademyTrust, EdubaseRoles.ROLE_BACKOFFICE)]
+        [InlineData((int) eLookupGroupType.ChildrensCentresGroup, EdubaseRoles.YCS)]
+        [InlineData((int) eLookupGroupType.ChildrensCentresCollaboration, EdubaseRoles.YCS)]
+        public void SetEditPermissions_GroupViewModel_IsLocalAuthorityEditable_False(int groupType, string role)
+        {
+            mockPrincipal.Setup(x => x.IsInRole(It.IsAny<string>())).Returns(false);
+            mockPrincipal.Setup(x => x.IsInRole(role)).Returns(true);
+            var viewModel = new GroupEditorViewModel()
+            {
+                GroupTypeId = groupType,
+                LinkedEstablishments = new GroupLinkedEstablishmentsViewModel() { Establishments = new List<EstablishmentGroupViewModel>() }
+            };
+
+            var result = GroupEditorViewModelRulesHandler.SetEditPermissions(viewModel, mockPrincipal.Object);
+
+            Assert.False(result.IsLocalAuthorityEditable);
+        }
```

Comment:
While I mentioned above that it's hard to spot missing test cases without going completely overboard, I do want to specifically call out that a logic branch/set of tests is missing/incomplete here, where zero linked establishments returns false for any user.

Overall this is a net improvement on the tests present previously though (thank you!) and personally I don't wish to block the PR -- feel free to mark this as resolved without code changes if desired.

Reply:
@CathLass any thoughts?

Reply:
yes, quite right - tests added (albeit after the refactoring was complete :(  )